src/Pmi/Order/Order.php

```diff
@@ -1174,17 +1175,31 @@ public function createOrderHistory($type, $reason = '')
             'order_id' => $this->order['id'],
             'user_id' => $this->app->getUser()->getId(),
             'site' => $this->app->getSiteId(),
-            'type' => $type,
+            'type' => $type === self::ORDER_REVERT ? self::ORDER_ACTIVE : $type,
             'created_ts' => new \DateTime()
         ];
         $ordersHistoryRepository = $this->app['em']->getRepository('orders_history');
         $status = false;
-        $ordersHistoryRepository->wrapInTransaction(function () use ($ordersHistoryRepository, $orderHistoryData, &$status) {
+        $ordersHistoryRepository->wrapInTransaction(function () use ($ordersHistoryRepository, $orderHistoryData, &$status, $type) {
             $id = $ordersHistoryRepository->insert($orderHistoryData);
             $this->app->log(Log::ORDER_HISTORY_CREATE, [
                 'id' => $id,
                 'type' => $orderHistoryData['type']
             ]);
+
+            //Get most recent order edit history id if exists while restoring or reverting an order
+            if ($type === self::ORDER_REVERT || $type === self::ORDER_RESTORE) {
+                $ordersQuery = "SELECT oh.id FROM orders_history oh WHERE oh.order_id = :orderId AND oh.type = :type ORDER BY oh.id DESC LIMIT 1";
+                $orderHistory = $this->app['em']->fetchAll($ordersQuery, [
```

Comment:
Is there a reason why you can't use the `fetchBy` method on `$ordersHistoryRepository`?

Reply:
Didn't realize that our fetchBy has a limit option. Thanks!